protocol.md

```diff
@@ -114,6 +114,114 @@ Observe the behavior of this algorithm in some failure scenarios. First, any of
+
+An alternative scenario is pairing in person.  The sharer has an app on their phone, and the helper might be a friend or family member that has an app on their phone. These can be unrelated apps, as long as both support the DeRec protocol. Each would launch their app and press a button to say that they want to pair with the other. The app might have two different buttons, to allow the user to pair as the sharer or as the helper. Or the app might have a simpler design, with a single pairing button, and then would actually run the protocol twice: once to pair as the sharer, and once to pair as the helper. That would allow two friends or family members to each be a helper to the other.
+
+For an in-person pairing, each phone might display a QR code containing the contact information. This just contains the user\'s encryption public key, internet communication information, and the nonce. If the pairing fails, then they can try again, and the QR code will have a different nonce. Either user could then use their phone camera to see the contact displayed by the other user. It doesn\'t matter whether the contact is being sent from sharer to helper or from helper to sharer. Either way, the protocol will allow them to pair.
+
+Once the contact information is transferred, the receiver of the contact sends the pairing request, and the other party replies with the pairing reply. The request includes the nonce to prevent Man in the Middle (MITM) attacks. For example if Alice uses her phone to see Bob\'s QR code, then Alice\'s phone will send Bob the pairing request, which contains the same nonce that was in the QR code. Since this nonce is randomly regenerated with each pairing attempt, a third party will not know it. So if a third party sent a pairing request to Bob, with the intent of pretending to be Alice, the attack would fail, because Bob\'s app would realize that the pairing request does not contain the same nonce that it had just displayed on the screen. Bob\'s app can therefore be confident that the pairing is occurring with the same human being that Bob just finished authenticating.
+
+It is important that the helper authenticate the sharer. If this is in person, then it should be someone they know, or someone they have authenticated from a driver license or passport.
+
+After the pairing is complete, that nonce is discarded. The sharer and helper apps each know the other\'s public keys for both encryption and signing, and each know the secret ID. This secret ID can be thought of as identifying the channel between them. From then on, the pair of (sharer public signing key, secret ID) will uniquely identify to the helper where a given share came from.
```

Comment:
> know the other\'s public keys for both encryption and signing

I presume we will use ephemeral:static ECDH? and include the ephemeral public key in the payload?

Reply:
See above comment. We can use a different term than "public key for encryption", if you think that's clearer. It just needs to convey that it is the public string of bits that are needed in order to send all future encrypted messages that can only be decrypted by the other party.